auto_tests/reconnect_test.c

```diff
@@ -0,0 +1,104 @@
+/* Auto Tests: Conferences.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#include "../testing/misc_tools.h"
+#include "../toxcore/friend_connection.h"
+#include "../toxcore/tox.h"
+#include "../toxcore/util.h"
+#include "check_compat.h"
+
+#define NUM_TOXES 2
+#define RECONNECT_TIME_MAX (FRIEND_CONNECTION_TIMEOUT + 3)
+
+typedef struct State {
+    uint32_t index;
+} State;
+
+#include "run_auto_test.h"
+
+static uint32_t tox_connected_count(uint32_t tox_count, Tox **toxes, State *state, uint32_t index)
+{
+    const size_t friend_count = tox_self_get_friend_list_size(toxes[index]);
+    uint32_t connected_count = 0;
+
+    for (size_t j = 0; j < friend_count; j++) {
+        if (tox_friend_get_connection_status(toxes[index], j, nullptr) != TOX_CONNECTION_NONE) {
```

Comment:
Ideally we'd use the state and friend connection callbacks for this, but maybe we need a bit more auto_test framework to make that less repetitive.

Reply:
Yes. I'm leaving it as is for now - quite a few of the tests use this deprecated function, best to fix them all at once.